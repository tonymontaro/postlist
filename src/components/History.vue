<template>
  <section>
    <h1 :class="headerStyle">List of actions committed</h1>
    <ul :class="ulStyle">
      <transition-group name="slide" class="w-full">
        <template v-for="historyItem in history">
          <HistoryItem
            :historyItem="historyItem"
            :key="historyItem.id"
            :replyHistory="replyHistory"
          />
        </template>
      </transition-group>
    </ul>
  </section>
</template>
<script>
import HistoryItem from "./HistoryItem";
import { BOX_STYLE } from "@/config";

export default {
  name: "actions",
  components: { HistoryItem },
  data() {
    return {
      headerStyle: [...BOX_STYLE, "text-xl"],
      ulStyle: [...BOX_STYLE, "p-5", "bg-gray-200"]
    };
  },
  computed: {
    history() {
      return this.$store.state.history.history;
    }
  },
  methods: {
    replyHistory(posts) {
      this.$store.commit("posts/replyHistory", posts);
    }
  }
};
</script>

<style scoped>
.slide-enter {
  opacity: 0;
}
.slide-enter-active {
  animation: slide-in 1s ease-out forwards;
  transition: opacity 0.5s;
}

@keyframes slide-in {
  from {
    transform: translateX(-20px);
  }
  to {
    transform: translateX(0);
  }
}

li {
  margin-bottom: 1px;
}
</style>
