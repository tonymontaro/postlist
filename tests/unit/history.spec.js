import Vuex from "vuex";
import { createLocalVue, mount, shallowMount } from "@vue/test-utils";

import History from "@/components/History";
import historyStore from "@/store/history";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("History Component", () => {
  let testHistoryItems;
  let store;
  let mutations;
  let wrapper;

  beforeEach(() => {
    testHistoryItems = [
      { id: 7, word: "Item 7 moved", prevState: [] },
      { id: 42, word: "Item 42 moved", prevState: [] },
      { id: 1, word: "Item 1 moved", prevState: [] },
      { id: 2, word: "Item 2 moved", prevState: [] }
    ];
    mutations = {
      "history/addAction": jest.fn(),
      "posts/replyHistory": jest.fn()
    };
    store = new Vuex.Store({
      state: {
        history: {
          history: testHistoryItems
        }
      },
      mutations
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test("renders history items", async () => {
    wrapper = shallowMount(History, {
      localVue,
      store,
      computed: {
        history: () => testHistoryItems
      }
    });
    expect(wrapper.findAll("historyitem-stub").length).toBe(4);
    expect(wrapper).toMatchSnapshot();
  });
});
