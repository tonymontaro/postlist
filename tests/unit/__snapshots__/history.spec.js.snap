// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Component renders history items 1`] = `
<section>
  <h1 class="p-3 py-5 bg-white shadow-lg rounded-sm white-box w-full text-xl">List of actions committed</h1>
  <ul class="p-3 py-5 bg-white shadow-lg rounded-sm white-box w-full p-5 bg-gray-200"><span name="slide" class="w-full"><historyitem-stub replyhistory="function () { [native code] }" historyitem="[object Object]"></historyitem-stub><historyitem-stub replyhistory="function () { [native code] }" historyitem="[object Object]"></historyitem-stub><historyitem-stub replyhistory="function () { [native code] }" historyitem="[object Object]"></historyitem-stub><historyitem-stub replyhistory="function () { [native code] }" historyitem="[object Object]"></historyitem-stub></span></ul>
</section>
`;
